### PR TITLE
Api workshift admin 5x

### PR DESCRIPTION
**Pull request by yasankarj**
Admin Workshift APIs

### CODE REVIEW

File: symfony/plugins/orangehrmAdminPlugin/Api/Model/WorkShiftModel.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * OrangeHRM Enterprise is a closed sourced comprehensive Human Resource Management (HRM)
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmAdminPlugin/Api/Model/WorkShiftModel.php:3]: pls remove the `Enterprise` code header .. we are still GPLv3 :)

File: symfony/plugins/orangehrmAdminPlugin/Api/Model/WorkShiftModel.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+use OrangeHRM\Core\Api\V2\Serializer\ModelTrait;
+use OrangeHRM\Core\Api\V2\Serializer\Normalizable;
+use OrangeHRM\Entity\WorkShift;
+
+class WorkShiftModel  implements Normalizable
+{
+    use ModelTrait;
+
+    public function __construct(WorkShift $workShift)
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmAdminPlugin/Api/Model/WorkShiftModel.php:32]: doc block pls

File: symfony/plugins/orangehrmAdminPlugin/Api/Model/WorkShiftModel.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            [
+                'id',
+                'name',
+                'hours_per_day',
+                'start_time',
+                'end_time'
+            ]
+        );
+    }
+}
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmAdminPlugin/Api/Model/WorkShiftModel.php:54]: missing end of line

File: symfony/plugins/orangehrmAdminPlugin/Api/WorkShiftAPI.php [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+    /**
+     * @param WorkShiftService $workShiftService
+     */
+    public function setWorkShiftService(WorkShiftService $workShiftService): void 
+    {
+        $this->workShiftService = $workShiftService;
+    }
+    
+    public function getAll(): EndpointResult
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmAdminPlugin/Api/WorkShiftAPI.php:65]: missing docblock

File: symfony/plugins/orangehrmAdminPlugin/Api/WorkShiftAPI.php [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    public function update(): EndpointResult 
+    {
+        // TODO: Implement update() method.
+    }
+
+    public function getValidationRuleForUpdate(): ParamRuleCollection 
+    {
+        // TODO: Implement getValidationRuleForUpdate() method.
+    }
+}
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmAdminPlugin/Api/WorkShiftAPI.php:131]: here too

File: symfony/plugins/orangehrmAdminPlugin/Api/WorkShiftAPI.php [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    {
+        return new ParamRuleCollection(
+            new ParamRule(
+                CommonParams::PARAMETER_ID,
+                new Rule(Rules::POSITIVE)
+            ),
+        );
+    }
+
+    public function create(): EndpointResult
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmAdminPlugin/Api/WorkShiftAPI.php:103]: pls add @todo's

File: symfony/plugins/orangehrmAdminPlugin/Dao/WorkShiftDao.php [hunk 7]
```diff
@@ -19,29 +19,73 @@
...
+        } catch (Exception $e) {
+            throw new DaoException($e->getMessage(), $e->getCode(), $e);
+        }
+    }
+
+    /**
+     * @param WorkShiftSearchFilterParams $workShiftSearchFilterParams
+     * @return Paginator
+     */
+    public function getWorkShiftListPaginator(WorkShiftSearchFilterParams $workShiftSearchFilterParams
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmAdminPlugin/Dao/WorkShiftDao.php:68]: this method signature against PSR standards 
```
public function getWorkShiftListPaginator (
 WorkShiftSearchFilterParams $workShiftSearchFilterParams
): Paginator {
```

File: symfony/plugins/orangehrmAdminPlugin/Dao/WorkShiftDao.php [hunk 8]
```diff
@@ -19,29 +19,73 @@
...
+     * @throws DaoException
+     */
+    public function getWorkShiftCount(WorkShiftSearchFilterParams $workShiftSearchFilterParams): int
     {
-        return $this->getRepository(WorkShift::class)->find($workShiftId);
+        try {
+            $paginator = $this->getWorkShiftListPaginator($workShiftSearchFilterParams);
+            return $paginator->count();
+        } catch (Exception $e) {
+            throw new DaoException($e->getMessage(), $e->getCode(), $e);
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmAdminPlugin/Dao/WorkShiftDao.php:87]: There is no use of catch-all exception and wrap as Dao exception  .. we will change this pattern globally

File: symfony/plugins/orangehrmAdminPlugin/Dto/WorkShiftSearchFilterParams.php [hunk 9]
```diff
@@ -0,0 +1,35 @@
+<?php
+/**
+ * OrangeHRM Enterprise is a closed sourced comprehensive Human Resource Management (HRM)
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmAdminPlugin/Dto/WorkShiftSearchFilterParams.php:3]: wrong code header

File: symfony/plugins/orangehrmAdminPlugin/Dto/WorkShiftSearchFilterParams.php [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+
+class WorkShiftSearchFilterParams  extends FilterParams
+{
+    public const ALLOWED_SORT_FIELDS = ['ws.name'];
+
+    public function __construct()
+    {
+        $this->setSortField('ws.name');
+    }
+}
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmAdminPlugin/Dto/WorkShiftSearchFilterParams.php:35]: here too

File: symfony/plugins/orangehrmAdminPlugin/entity/Decorator/WorkShiftDecorator.php [hunk 11]
```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * OrangeHRM Enterprise is a closed sourced comprehensive Human Resource Management (HRM)
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmAdminPlugin/entity/Decorator/WorkShiftDecorator.php:3]: wrong code header

File: symfony/plugins/orangehrmAdminPlugin/entity/Decorator/WorkShiftDecorator.php [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+    public function getStartTime()
+    {
+        return $this->getDateTimeHelper()->formatDateTimeToTimeString($this->getWorkShift()->getStartTime());
+    }
+    
+    public function getEndTime()
+    {
+        return $this->getDateTimeHelper()->formatDateTimeToTimeString($this->getWorkShift()->getEndTime());
+    }
+}
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmAdminPlugin/entity/Decorator/WorkShiftDecorator.php:62]: here too

**Comment by RajithaKumara** [comment]
Closing this since commits moved to #940
@VageeshOrangeHRM can you address these feedback in #940

**Pull request closed without merging**